addons/pos_adyen/models/pos_payment_method.py

```diff
@@ -52,6 +49,14 @@ def _check_adyen_terminal_identifier(self):
                                              company=existing_payment_method.company_id.name,
                                              payment_method=existing_payment_method.display_name))
 
+    @api.onchange('use_payment_terminal')
+    def _onchange_use_payment_terminal(self):
+        super()._onchange_use_payment_terminal()
+        if self.use_payment_terminal == 'adyen' and not self.adyen_api_key:
+            existing_payment_method = self.search([('use_payment_terminal', '=', 'adyen'), ('adyen_api_key', '!=', False)], limit=1)
+            if existing_payment_method:
+                self.adyen_api_key = existing_payment_method.adyen_api_key
+
```

Comment:
could you take this into account?

![image](https://github.com/user-attachments/assets/212a1f89-bfc9-4f63-852b-34ddd0d13339)

[reference](https://www.odoo.com/documentation/master/developer/reference/backend/orm.html#:~:text=will%20be%20ignored-,Danger,-Since)